src/sampleutil.cpp

```diff
@@ -294,72 +333,71 @@ void SampleUtil::deinterleaveBuffer(CSAMPLE* pDest1, CSAMPLE* pDest2,
 // static
 void SampleUtil::linearCrossfadeBuffers(CSAMPLE* pDest,
         const CSAMPLE* pSrcFadeOut, const CSAMPLE* pSrcFadeIn,
-        unsigned int iNumSamples) {
+        int iNumSamples) {
     const CSAMPLE_GAIN cross_inc = CSAMPLE_GAIN_ONE
             / CSAMPLE_GAIN(iNumSamples / 2);
-    CSAMPLE_GAIN cross_mix = CSAMPLE_GAIN_ZERO;
-    for (unsigned int i = 0; i + 1 < iNumSamples; i += 2) {
-        pDest[i] = pSrcFadeIn[i] * cross_mix
-                + pSrcFadeOut[i] * (CSAMPLE_GAIN_ONE - cross_mix);
-        pDest[i + 1] = pSrcFadeIn[i + 1] * cross_mix
-                + pSrcFadeOut[i + 1] * (CSAMPLE_GAIN_ONE - cross_mix);
-        cross_mix += cross_inc;
```

Comment:
The loop counter must be ++i only, do be vectorized. i += 2 prevents vectorizing.